### PR TITLE
Add unit tests in Kinesis consumer

### PR DESCRIPTION
**Pull request by KKcorps**
The PR adds unit tests for new Kinesis consumers for Apache pinot. The work for integration tests is still in progress.

### CODE REVIEW

File: pinot-plugins/pinot-stream-ingestion/pinot-kinesis/pom.xml [hunk 1]
```diff
@@ -36,9 +36,26 @@
   <properties>
     <pinot.root>${basedir}/../../..</pinot.root>
     <phase.prop>package</phase.prop>
-    <aws.version>2.15.50</aws.version>
+    <aws.version>2.14.28</aws.version>
+    <powermock.version>2.0.9</powermock.version>
```
Comment (npawar) [inline pinot-plugins/pinot-stream-ingestion/pinot-kinesis/pom.xml:40]: remove the powermock if it is not being used anymore

File: pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisStreamMetadataProvider.java [hunk 2]
```diff
@@ -83,7 +91,7 @@ public long fetchPartitionOffset(@Nonnull OffsetCriteria offsetCriteria, long ti
     List<PartitionGroupInfo> newPartitionGroupInfos = new ArrayList<>();
 
     Map<String, Shard> shardIdToShardMap =
-        _kinesisConnectionHandler.getShards().stream().collect(Collectors.toMap(Shard::shardId, s -> s));
+        _kinesisConnectionHandler.getShards().stream().collect(Collectors.toMap(Shard::shardId, s -> s, (s1, s2) -> s1));
```
Comment (npawar) [inline pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisStreamMetadataProvider.java:94]: why would a collision happen here?

**Review by npawar — APPROVED** [review]

File: pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisStreamMetadataProvider.java (hunk 2 above)
Comment (KKcorps) [reply pinot-plugins/pinot-stream-ingestion/pinot-kinesis/src/main/java/org/apache/pinot/plugin/stream/kinesis/KinesisStreamMetadataProvider.java:94]: I also wasn't able to figure this out but while running a collision was definitely happening and error was being thrown.

File: pinot-plugins/pinot-stream-ingestion/pinot-kinesis/pom.xml (hunk 1 above)
Comment (KKcorps) [reply pinot-plugins/pinot-stream-ingestion/pinot-kinesis/pom.xml]: resolved.

**Pull request merged**